### ISSUE TITLE
`ins_serial_no` requirement negates use of `hfx_deployments.csv` as a test data set

### ISSUE THREAD

**Issue by mhpob** [ocean-tracking-network/glatos#202]
The addition of `ins_serial_no` as a required column in `validate_glatos_receivers` seems to have broken `test-read_otn_deployments` (commit 5f6b727). Seems to run through `glatos_check_col_names`, also introduced in 5f6b727.

The test data set, `hfx_deployments.csv`, does not contain an `ins_serial_no` column.

**Comment by chrisholbrook** [comment]
I am not aware of any glatos function that requires `ins_serial_no`, so perhaps this requirement should be relaxed; but it also seems like a reasonable requirement for an object about devices. @jdpye, is [hfx_deployments.csv](https://github.com/ocean-tracking-network/glatos/blob/main/inst/extdata/hfx_deployments.csv) still an active file structure in OTN?

**Comment by jdpye** [comment]
on further review, the existence of columns like off_set in this example means it was likely to be sourced from our longer-form metadata sheets, where we track things like how far up the riser is from the bottom mooring.

 I still think we should accept ins_serial_no and agree to looking into whether we can soften its requirement, but this HFX example is even more than our basic deployment metadata requires.

**Comment by jdpye** [comment]
further-further review - the true source looks to be our geoserver layer otn.stations_receivers. We're looking into if we can add `ins_serial_no` to that layer. https://members.oceantrack.org/geoserver/otn/ows?service=WFS&version=1.0.0&request=GetFeature&typeName=otn:stations_receivers&outputFormat=csv

**Comment by jdpye** [comment]
Instrument serial number is marked 'required' on our deployment sheets, we usually follow up or pull from manufacturer specifications if we don't get them from the data submitters. So we will be in good shape to require this, just have to express it in the datasets we're pushing out into the world via GeoServer.

https://ocean-tracking-network.github.io/node-manager-training/07_deploy_metadata/index.html

**Comment by mhpob** [comment]
@jdpye it seems where we left this was to wait for OTN to express `in_serial_no` on the geoserver. Has this been done?

**Comment by jdpye** [comment]
yep, the field on the layer is `rcvr_serial` and we should be properly populating both that and the `map_code` field now.

**Comment by ricardd** [comment]
@chrisholbrook, @jdpye and @mhpob, is there a resolution in sight for this issue?

An error still occurs when trying to load a CSV file from OTN (the file downloaded when clicking on "Deployments" from an OTN project page), and the error goes away after I add a column called `ins_serial_no` to the CSV.

<img width="715" height="258" alt="Image" src="https://github.com/user-attachments/assets/8cf8a26d-f03c-4704-8e23-8bc1f8516717" />

The CSV file I am using in the example above is from this project: 
https://members.oceantrack.org/project?ccode=CBS

**Comment by mhpob** [comment]
This issue was originally just talking about example data we're testing the package against, so thanks for flagging a re-world impact. @ricardd FWIW the function is operating correctly as designed.

@jdpye is this an "old data" vs "new data" issue? If so, @chrisholbrook, and if this is not a GLATOS vs OTN issue, should this requirement be relaxed for backwards compatibility? Perhaps with a warning?

**Comment by ricardd** [comment]
@mhpob I don't think that it's an "old data" vs "new data" issue. I had an error using `read_otn_detections` as well, but it disappeared once I downloaded the detections again, and am assuming that the deployments file I am using is in "new data" format..

**Comment by chrisholbrook** [comment]
@ricardd My preference would be to keep the constraint in place by default (per our earlier decision on this issue) but allow validation to be skipped (i.e., allow `validate = FALSE` to pass from `read_otn_deployments()` to `as_glatos_receivers()`). This would not be strictly backward compatible because earlier code/data would still break (those breaking changes have already been released); but this would allow access to earlier functionality (validate_* functions were added in glatos 0.8.0).

@jdpye @mhpob Please chime in if you see any issues. Either @haydento or I will submit a PR on this as a soon as we are able.

**Comment by haydento** [comment]
Hey @jdpye I think this is nearly fixed.  Any chance your shop could take a quick look at `read_otn_deployments` and `read_otn_detections` in dev?  Unit tests  are not passing/implemented but I think the functions are mostly working?
